war/src/main/js/add-item.js

```diff
@@ -16,378 +16,197 @@ var jRoot = $('head').attr('data-rooturl');
-        if ($nameInput.val() === '') {
-          $nameInput.focus();
+        $this.find('input[type="radio"][name="mode"]').prop('checked', true);
+        $('input[type="text"][name="from"]', '#createItem').val('');
+        cleanValidationMessages('.add-item-copy');
+        if (isItemNameEmpty()) {
+          activateValidationMessage('#itemname-required', '.add-item-name');
+          $('input[name="name"][type="text"]', '#createItem').focus();
         }
       }
-
       $item.click(setSelectState);
 
       return $item;
     }
 
-    function drawIcon(elem){
-      var $icn = $('<div class="icn">');
-      if (!elem.iconFilePathPattern) {
-        var colors = ['c-49728B','c-335061','c-D33833','c-6D6B6D','c-DCD9D8','other'];
+    function drawIcon(elem) {
+      var $icn;
+      if (elem.iconFilePathPattern) {
+        $icn = $('<div class="icon">');
+        var iconFilePath = jRoot + '/' + elem.iconFilePathPattern.replace(":size", "48x48");
+        $(['<img src="', iconFilePath, '">'].join('')).appendTo($icn);
+      } else {
+        $icn = $('<div class="default-icon">');
+        var colors = ['c-49728B','c-335061','c-D33833','c-6D6B6D', 'c-6699CC'];
         var desc = elem.description || '';
         var name = elem.displayName;
-        var colorClass= colors[(desc.length) % 6];
+        var colorClass= colors[(desc.length) % 4];
         var aName = name.split(' ');
         var a = name.substring(0,1);
-        var b = ((aName.length === 1)?
-            name.substring(1,2):
-              aName[1].substring(0,1));
-        $([
-          '<span class="dfIcn"><span class="a">',a,'</span><span class="b">',b,'</span></span>'
-         ].join(''))
-          .appendTo($icn);
-        return $icn.addClass('df').addClass(colorClass);
+        var b = ((aName.length === 1) ? name.substring(1,2) : aName[1].substring(0,1));
+        $(['<span class="a">',a,'</span><span class="b">',b,'</span>'].join('')).appendTo($icn);
+        $icn.addClass(colorClass);
       }
-
-      var iconFilePath = jRoot + '/' + elem.iconFilePathPattern.replace(":size", "48x48");
-      $(['<span class="img" style="background:url(', iconFilePath, ')"></span>'].join(''))
-        .appendTo($icn);
-
       return $icn;
     }
 
-    var categoriesWithCopy = addCopyOption(data.categories);
+    // The main panel content is hidden by default via an inline style. We're ready to remove that now.
+    $('#add-item-panel').removeAttr('style');
+
+    // Render all categories
+    var $categories = $('div.categories');
+    $.each(data.categories, function(i, elem) {
+      drawCategory(elem).appendTo($categories);
+    });
+
+    // Focus
+    $("#add-item-panel").find("#name").focus();
+
+    // Init NameField
+    $('input[name="name"]', '#createItem').blur(function() {
+      if (!isItemNameEmpty()) {
+        var itemName = $('input[name="name"]', '#createItem').val();
+        $.get(jRoot + "/checkJobName", { value: itemName }).done(function(data) {
```

Comment:
While `/checkJobName` should continue to work (since the `StaplerFallback` for `Jenkins` is its `primaryView`), you should now be able to just append `checkJobName` to either the `ItemGroup`’s URL (by way of `StaplerFallback`):

* `/checkJobName`
* `/job/somefolder/checkJobName` (though prior to https://github.com/jenkinsci/cloudbees-folder-plugin/pull/51 this will give slightly inaccurate results)

or to the `View`’s URL:

* `/view/All/checkJobName`
* `/job/somefolder/view/All/checkJobName`